### DISCUSSION TITLE
Physical Button For Pump Control

### DISCUSSION THREAD

**Discussion by coolboarder** [tagyoureit/nodejs-poolController#860] [Q&A]
I'm looking for  a way to have a physical button to toggle the pump on and off.  Is there a way in nodejs-pool to handle a button input and activate/deactivate the pool circuit?  As a bonus I would like to control LEDs to indicate pump status and heater status.

Description of my setup: I have an basic pool with just a pump and a gas heater.  No spa, no electronic valves nothing else.  Pump is controlled by an old analog dial timer.  

I am going to install nodejs-pool as a stand alone Nixie controller to control the pump and heater.  Hardware consists of a 20v dc power supply, Raspberry PI 4, Sequent BAS hat, 3 relay hat, pump relay and 10k temp probe.  Here is a picture of the setup on the bench.

![PoolController](https://github.com/tagyoureit/nodejs-poolController/assets/33079415/2622d5d4-c9c9-4dc3-b494-a9ae79604dc4)

**Comment by rock-crusher** [comment]
Hi @coolboarder,
I’m really surprised you haven’t received a response yet.
@tagyoureit & @rstrouse are usually always very good at responding to these type questions, but hey we all have lives to live.
I can’t count how many times they have come to my rescue with all the idiotic questions I have asked over the years. 
Thank you once again @tagyoureit & @rstrouse for all the time you spent with me working on getting my project up and running!   
Anyway, the link below may or may not be exactly what you’re looking for, but I think it will at least get you going in the right direction to achieve your goal.
Have fun & good luck!
https://github.com/tagyoureit/nodejs-poolController/discussions/508

**Reply by rstrouse** [reply]
Just been busy.  However, @rock-crusher  is correct.  You can feed any signal from REM back to njsPC to trigger an event.  For your LED display you can set a trigger that turns it on when the pump.isOn == true.

**Reply by mhmatttu** [reply]
I raised a similar question and issue in the [#508](https://github.com/tagyoureit/nodejs-poolController/discussions/508#discussioncomment-9600396) thread but found this one too.  @coolboarder Did you. Have any luck getting this to work with MEGA-BAS inputs?

**Reply by rstrouse** [reply]
I have added a toggle endpoint to gpio and BAS inputs.  You should now be able to toggle with a pushbutton.
